py/strato/common/log/log2text.py

```diff
@@ -227,20 +230,76 @@ def printLogs(logFiles, formatter):
 
 
 def updateConfFile(field, value):
-    with open(LOG_CONFIG_FILE_PATH, 'r') as f:
-        conf = yaml.load(f.read())
+    try:
+        with open(LOG_CONFIG_FILE_PATH, 'r') as f:
+            conf = yaml.load(f.read())
+    except:
+        print "No configuration file was found. creating new"
+        conf = {}
+    finally:
         conf[field] = value
     with open('/tmp/stratolog-conf.tmp', 'w') as f:
         f.write(yaml.dump(conf))
+        f.close()
     os.system('sudo mv /tmp/stratolog-conf.tmp %s' % LOG_CONFIG_FILE_PATH)
 
 
-def universalOpen(filePath, mode='r'):
+def universalOpen(filePath, mode='r', tail=0):
     if filePath.endswith('.gz'):
-        return gzip.GzipFile(filePath, mode)
+        fileObj = gzip.GzipFile(filePath, mode)
     else:
-        return open(filePath, mode)
+        if tail > 0:
+            fileObj = tailFile(filePath, tail)
+        else:
+            fileObj = open(filePath, mode)
+    return fileObj
+
+def tailFile(filePath, n):
+    p = subprocess.Popen(["tail", "-n", str(n), filePath], stdout=subprocess.PIPE)
+    return p.stdout
+
+def resolveHostname(host):
+    try:
+        socket.gethostbyname(host)
+        return True
+    except:
+        return False
+
+def runRemotely(host, ignoreArgs):
+    try:
+        configFile = yaml.load(open(LOG_CONFIG_FILE_PATH, 'r').read())
+    except:
+        configFile = {}
+        print "Configuration file was not found"
+
+    args = [arg for arg in sys.argv[1:] if arg not in ignoreArgs]
+    command = "strato-log %s" % ' '.join(args)
 
+    user = configFile.get("defaultRemoteUser", 'root')
+    password = configFile.get("defaultRemotePassword", 'password')
+
+    possibleResolveSuffixes = configFile.get("possibleResolveSuffixes", [])
+    hostname=findHostname(host, possibleResolveSuffixes)
+
+    host = "%(user)s@%(hostname)s" % dict(user=user,
+                                          hostname=hostname)
+
+    sshCommand = 'sshpass -p %(password)s ssh -o ServerAliveInterval=5 -o ServerAliveCountMax=1 -o ' \
+                 'StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null %(host)s %(command)s | less -r'
+
+    os.system(sshCommand % dict(command=command,
+                                host=host,
+                                password=password))
+
+def findHostname(host, possibleResolveSuffixes):
+    if resolveHostname(host):
+        return host
+    else:
+        for suffix in possibleResolveSuffixes:
+            if resolveHostname(host + suffix):
+                return host + suffix
+    print "No reachable host was found for %s" % host
+    exit(1)
```

Comment:
sure, but better to have single entry and exit point for you code. It's much easier to read.
Noone should guess that exit is called internally.